pyhealth/datasets/splitter.py

```diff
@@ -40,9 +36,9 @@ def split_by_visit(
         int(len(dataset) * ratios[0]) : int(len(dataset) * (ratios[0] + ratios[1]))
     ]
     test_index = index[int(len(dataset) * (ratios[0] + ratios[1])) :]
-    train_dataset = torch.utils.data.Subset(dataset, train_index)
-    val_dataset = torch.utils.data.Subset(dataset, val_index)
-    test_dataset = torch.utils.data.Subset(dataset, test_index)
+    train_dataset = SampleSubset(dataset, train_index) # type: ignore
+    val_dataset = SampleSubset(dataset, val_index) # type: ignore
+    test_dataset = SampleSubset(dataset, test_index) # type: ignore
```

Comment:
torch native Subset does not support iterable dataset, we use a custom-defined iterable dataset here.